src/components/badge/index.tsx

```diff
@@ -132,10 +136,12 @@ class Badge extends PureComponent<BadgeProps> {
   }
 
   getAccessibilityProps() {
-    const {onPress, icon, label} = this.props;
-
+    const {onPress, icon, label, accessibilityLabel} = this.props;
     return {
-      accessibilityLabel: icon ? 'badge' : label ? `${label} new items` : undefined,
+      accessibilityLabel: accessibilityLabel ? accessibilityLabel :
+        label ? `${label} new items` :
+        icon ? 'badge' :
+        undefined,
```

Comment:
couple of things.. 
First, when you do `x ? x : y` it's clean to do `x || y`
So in this case I'd would `accessibilityLabel || <fallback logic>` 

and the other thing, we always want to fallback to `badge` as an accessibility label. 
So whether it's an icon badge or empty badge we'd like to use `badge` for its a11y label.